src/Model/IModel.ts

```diff
@@ -36,6 +37,7 @@ export interface IModel<T extends IModelData> extends IModelMeta<T>, IModelData,
   initializeNewModel(): void;
   markForDestruction(): void;
   parseFieldValue(fieldName: string, value: any): Promise<any>;
+  getServiceForRelationship(relationshipKey: string): DataService<any>;
```

Comment:
Should we type this with a `T`?

Reply:
Unfortunately not, because that implies it's the DataService associated to the current model, when in fact it's the DataService associated to the related model, which could be any DataService.